modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -664,16 +824,91 @@ public void testPasswordHistoryWithModifiedEncryption() throws Exception {
 					portalException.getClass());
 
 				Assert.assertEquals(
-					"{SHA-384}66b6aa56af08dc8caf7e001683058338244f436de61d40" +
-						"e342d0c69bda9f73cd6d167fdb29925db579923bdcef1fe5ae",
-					user.getPassword());
+					Authenticator.SUCCESS,
+					_userLocalService.authenticateByEmailAddress(
+						user.getCompanyId(), user.getEmailAddress(), password2,
+						null, null, null));
 			}
 			finally {
 				ServiceContextThreadLocal.popServiceContext();
 			}
 		}
 	}
 
+	@Test
+	public void testPasswordHistoryWithModifiedEncryption() throws Exception {
+		try (AutoCloseable autoCloseable1 =
+				ReflectionTestUtil.setFieldValueWithAutoCloseable(
+					DigesterImpl.class, "_BASE_64", false);
+			AutoCloseable autoCloseable2 =
+				ReflectionTestUtil.setFieldValueWithAutoCloseable(
+					PasswordEncryptorUtil.class,
+					"_PASSWORDS_ENCRYPTION_ALGORITHM", "SHA-384")) {
+
+			User user = UserTestUtil.addUser();
+
+			try (SafeCloseable safeCloseable =
+					_updateDefaultPasswordPolicyWithSafeCloseable(
+						passwordPolicy -> {
+							passwordPolicy.setHistory(true);
+							passwordPolicy.setHistoryCount(2);
+						})) {
+
+				String password1 = "password1";
+				String password2 = "password2";
+
+				try {
+					ServiceContextThreadLocal.pushServiceContext(
+						ServiceContextTestUtil.getServiceContext(
+							user.getGroupId(), user.getUserId()));
+
+					user = _userLocalService.updatePassword(
+						user.getUserId(), password1, password1, false, false);
+
+					Assert.assertEquals(
+						"{SHA-384}f5e2dd85fe11cec4c913f0f1fcecddb4a654dd92852" +
+							"f978d6345638a0779a5e77ea39d33d6254bde0e1afa7a6c8" +
+								"ef0b9",
+						user.getPassword());
+
+					user = _userLocalService.updatePassword(
+						user.getUserId(), password2, password2, false, false);
+
+					Assert.assertEquals(
+						"{SHA-384}66b6aa56af08dc8caf7e001683058338244f436de61" +
+							"d40e342d0c69bda9f73cd6d167fdb29925db579923bdcef1" +
+								"fe5ae",
+						user.getPassword());
+
+					Assert.assertEquals(
+						Authenticator.SUCCESS,
+						_userLocalService.authenticateByEmailAddress(
+							user.getCompanyId(), user.getEmailAddress(),
+							password2, null, null, null));
+
+					_userLocalService.updatePassword(
+						user.getUserId(), password1, password1, false, false);
+
+					Assert.fail();
+				}
+				catch (PortalException portalException) {
+					Assert.assertEquals(
+						UserPasswordException.MustNotBeRecentlyUsed.class,
+						portalException.getClass());
+
+					Assert.assertEquals(
+						"{SHA-384}66b6aa56af08dc8caf7e001683058338244f436de61" +
+							"d40e342d0c69bda9f73cd6d167fdb29925db579923bdcef1" +
+								"fe5ae",
+						user.getPassword());
+				}
```

Comment:
Catch the specific exception (MustNotBeRecentlyUsed) instead of asserting its type; let any other exn cause a test error.